### DISCUSSION TITLE
tomcat support

### DISCUSSION THREAD

**Discussion by zhuanshenlikai** [eclipse-theia/theia#12449] [Q&A]
I try to run java project with tomcat in Theia, but i do not see any supports for tomcat. I also search the plugins for tomcat in open-vsx, but I still haven’t found what I want. Is there any way to run java web project with tomcat in Theia?

**Comment by msujew** [comment]
Hey @zhuanshenlikai,

there doesn't seem to be any dedicated tomcat extensions on the open-vsx marketplace. The vscode marketplace seems to offer [this](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-community-server-connector) and [this](https://marketplace.visualstudio.com/items?itemName=adashen.vscode-tomcat). You might need to ask the developers of these extensions whether they're willing to host them on open-vsx as well.

Theia doesn't offer any framework/programming language specific functionalities aside from the vscode extension host.
